crates/turborepo-lib/src/formatter.rs

```diff
@@ -0,0 +1,103 @@
+use std::marker::PhantomData;
+
+use chrono::Local;
+use owo_colors::{
+    colors::{Black, Default, Red, Yellow},
+    Color, OwoColorize,
+};
+use tracing::{field::Visit, Event, Level, Subscriber};
+use tracing_subscriber::{
+    fmt::{format::Writer, FmtContext, FormatEvent, FormatFields},
+    registry::LookupSpan,
+};
+
+/// The formatter for TURBOREPO
+///
+/// This is a port of the go formatter, which follows a few main rules:
+/// - Errors are red
+/// - Warnings are yellow
+/// - Info is default
+/// - Debug and trace are default, but with timestamp and level attached
+///
+/// This formatter does not print any information about spans, and does
+/// not print any event metadata other than the message (which is set
+/// when you call `debug!(...)` or `info!(...)` etc.
+pub struct TurboFormatter;
+
+impl<S, N> FormatEvent<S, N> for TurboFormatter
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(
+        &self,
+        _ctx: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> std::fmt::Result {
+        // Format values from the event's's metadata:
+        let metadata = event.metadata();
+        match metadata.level() {
+            &Level::ERROR => write!(
+                writer,
+                "{} ",
+                metadata.level().to_string().bg::<Red>().fg::<Black>()
+            )
+            .and_then(|_| write_message::<Red, Default>(writer, event)),
+            &Level::WARN => write!(
+                writer,
+                "{} ",
+                metadata.level().to_string().bg::<Yellow>().fg::<Black>(),
+            )
+            .and_then(|_| write_message::<Yellow, Default>(writer, event)),
+            &Level::INFO => write_message::<Default, Default>(writer, event),
+            // trace and debug use the same style
+            _ => {
+                let now = Local::now();
+                write!(
+                    writer,
+                    "{} [{}] {}: ",
+                    // build our own timestamp to match the hashicorp/go-hclog format used by the
+                    // go                 // binary
```

Comment:
Nice. Make sure these match the go binary when running with `-vvv` (or similar)!

Also, weird comment format here (trailing `// binary`)